sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -1069,9 +1069,13 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
   }
 
   std::lock_guard<std::mutex> Lock{*PiESimdSurfaceMapLock};
-  assert(PiESimdSurfaceMap->find((*RetMem)->SurfaceIndex) ==
-             PiESimdSurfaceMap->end() &&
-         "Failure from CM-managed buffer creation");
+  if (PiESimdSurfaceMap->find((*RetMem)->SurfaceIndex) !=
+      PiESimdSurfaceMap->end()) {
+    if (PrintPiTrace) {
+      std::cerr << "Failure from CM-managed buffer creation" << std::endl;
+    }
```

Comment:
This code patterns is used at least 8 times. In my opinion, it deserves a function:
```suggestion
    printPiTrace("Failure from CM-managed buffer creation");
```
where printPiTrace() is a func accepting std::string or 'const char *' and that checks 'PrintPiTrace' before writing the argument to std::cerr.

Reply:
Done.